PowerShell/ScubaGear/Testing/Unit/PowerShell/Orchestrator/Invoke-ScubaConfig.Tests.ps1

```diff
@@ -0,0 +1,142 @@
+                }
+                
+                # Credentiasl test:  pass the credetails as an argument combined with a
+                # config file that does not have them
+                function CredsTest() {
+                    $pass = $true
+                    Invoke-Scuba @SplatParamsCreds
+                    $ConfTestCreds = [ScubaConfig]::GetInstance().Configuration
+
+                    # Scuba config should now have the resultes with auth params
+                    # Verify that all values are present and match
+                    if ( (Compare-Object @($ScubaTestCreds.keys) @($ScubaConfRef.keys)))
+                    {
+                        $pass = $false
+                    }
+                    foreach ($key in $ConfTestCreds.keys )
+                    {
+                         if (( Compare-Object  $ConfTestCreds.$key  $ScubaConfRef.$key ) -ne $none )
+                         {
+                            $pass = $false
+                         }
+                    }
+                    return $pass
+                }
+            }
+
+            It "Verify overide parameter ""<parameter>"" with value ""<value>""" -ForEach @(
+                @{ Parameter = "M365Environment";       Value = "gcc"                           }
+                @{ Parameter = "M365Environment";       Value = "commercial"                    }
+                @{ Parameter = "ProductNames";          Value = "teams"                         }
+                @{ Parameter = "ProductNames";          Value = @("teams","aad")                }
+                @{ Parameter = "OPAPath";               Value = ".."                            }
+                @{ Parameter = "Login";                 Value = $false                          }
+                @{ Parameter = "DisconnectOnExit";      Value = $true                           }
+                @{ Parameter = "OutPath";               Value = ".."                           }
+                @{ Parameter = "OutFolderName";         Value = "M365BaselineConformance_mod"   }
+                @{ Parameter = "OutProviderFileName";   Value = "ProviderSettingsExport_mod"    }
+                @{ Parameter = "OutRegoFileName";       Value = "TestResults_mod"               }
+                @{ Parameter = "OutReportName";         Value = "BaselineReports_mod"           }
+                @{ Parameter = "Organization";          Value = "mod.sub.domain.com"            }
+                @{ Parameter = "AppID";                 Value = "0123456789badbad"              }
+                @{ Parameter = "CertificateThumbprint"; Value = "BADBAD9786543210"              }
+                ){
+                    OverrideTest $Parameter $Value | Should -Be $true
+            }
+
+            It "Verify credentials passed in that are not in config file" {
+                Invoke-Scuba @SplatParamsCreds
+                $ConfTestCreds = [ScubaConfig]::GetInstance().Configuration
```

Comment:
If I add this line should the test pass?  
```suggestion
                Invoke-Scuba @SplatParamsCreds
                $ConfTestCreds.AppID | Should -Not -BeExactly $ScubaConfRef.AppID
                $ConfTestCreds = [ScubaConfig]::GetInstance().Configuration
```

Reply:
Not sure I follow this.  If I run the pester tests it passes ( cd to the Orxchestrator dir and run )

Reply:
I see this is happening from the pipeline ( difference from command line test ).  I did not know the pipeline was more sensitive

Reply:
I have restored the local function definition as per the recommendation cited in the commit 
https://www.logitblog.com/mastering-mocking-with-pester-commandnotfoundexception/
Think this is the best way to go.  Unit tests now pass OK in the pipeline